addons/binding/org.openhab.binding.dsmr/src/main/java/org/openhab/binding/dsmr/internal/device/DSMRPort.java

```diff
@@ -0,0 +1,498 @@
+
+                // Configure Serial Port based on specified port speed
+                logger.debug("Configure serial port parameters: {}", portSettings);
+
+                if (portSettings != null) {
+                    serialPort.setSerialPortParams(portSettings.getBaudrate(), portSettings.getDataBits(),
+                            portSettings.getStopbits(), portSettings.getParity());
+
+                    // SerialPort is ready, open the reader
+                    logger.debug("SerialPort opened successful");
+                    try {
+                        bis = new BufferedInputStream(serialPort.getInputStream());
+                    } catch (IOException ioe) {
+                        logger.error("Failed to get inputstream for serialPort. Closing port", ioe);
+
+                        dsmrPortListener.handleDSMRPortEvent(DSMRPortEvent.ERROR);
+                    }
+                    logger.info("DSMR Port opened successful");
+                    portStatus.isOpen = true;
+
+                    try {
+                        serialPort.addEventListener(this);
+                    } catch (TooManyListenersException tmle) {
+                        logger.error("DSMR binding will not be operational.", tmle);
+                    }
+
+                    /*
+                     * Let the listener known the port is opened
+                     * before we start listening for serial port events
+                     */
+                    dsmrPortListener.handleDSMRPortEvent(DSMRPortEvent.OPENED);
+
+                    /*
+                     * Start listening for events
+                     */
+                    serialPort.notifyOnDataAvailable(true);
+                    serialPort.notifyOnBreakInterrupt(true);
+                    serialPort.notifyOnFramingError(true);
+                    serialPort.notifyOnOverrunError(true);
+                    serialPort.notifyOnParityError(true);
+
+                    // The binding is ready, let the meter know we want to receive values
+                    serialPort.setRTS(true);
+                } else {
+                    logger.error("Invalid port parameters, closing port:{}", portSettings);
+
+                    dsmrPortListener.handleDSMRPortEvent(DSMRPortEvent.CONFIGURATION_ERROR);
+                }
+            } catch (NoSuchPortException nspe) {
+                logger.error("Port {} does not exists", portName, nspe);
+
+                dsmrPortListener.handleDSMRPortEvent(DSMRPortEvent.DONT_EXISTS);
+            } catch (PortInUseException piue) {
+                logger.error("Port already in use: {}", portName, piue);
+
+                dsmrPortListener.handleDSMRPortEvent(DSMRPortEvent.IN_USE);
+            } catch (UnsupportedCommOperationException ucoe) {
+                logger.error(
+                        "Port does not support requested port settings " + "(invalid dsmr:portsettings parameter?): {}",
```

Comment:
Can you compare your log levels with the guidelines, these should be more like warnings or given that you also change the thing status maybe nothing at all.

See [E.4 @ the coding guidelines](http://docs.openhab.org/developers/development/guidelines.html#e-logging)